flink-table/flink-table-uber/pom.xml

```diff
@@ -88,6 +88,11 @@ under the License.
 			<artifactId>flink-cep</artifactId>
 			<version>${project.version}</version>
 		</dependency>
+		<dependency>
+			<groupId>org.apache.flink</groupId>
```

Comment:
Why do we again include the file connector in the uber-jar?

Reply:
This PR doesn't changes the uber jar building, which on master already includes flink-connector-files. The reason for this specific change is that, before this PR, flink-table-common depends on flink-connector-files, so flink-table-uber depends on flink-connector-files through flink-table-common. Now none of the packages in flink-table-* depends anymore on flink-connector-files, so that's why we need this change.

Reply:
I'm also skeptical about this change. It would be better to explicitly add `flink-connector-files` as a separate dependency to `lib`. It must not be part of the the table jars.

Reply:
We should also add a comment in the docs that the filesystem connector is in `lib` by default.

Reply:
> I'm also skeptical about this change

But this is no different from master. If you build flink-table-uber in master, you'll see that flink-connector-files is inside the resulting jar, because this is brough in transitively from flink-table-common. If we want to do such change, I suggest we do it in a separate PR with a separate issue as it's user facing.

Reply:
Hmm, but when reading the ticket https://issues.apache.org/jira/browse/FLINK-24687 isn't that exactly what we want to solve? We want to exclude the `flink-connector-files` from the uber jar to prevent conflicts with user jars if we keep including it we did not fix the problem.

Reply:
Uhm you're right, I forgot what the initial issue title was about :smiley:

Reply:
I've pushed a commit to remove the flink-connector-files shading from the table-uber and to add the flink-connector-files to the lib folder in the distribution. I've also manually tested the new distribution output with a sample source - sink sql batch job using the sql client. 

This is the distribution tree now:

```
!2174 ➜ tree
.
├── bin
│   ├── bash-java-utils.jar
│   ├── config.sh
│   ├── find-flink-home.sh
│   ├── flink
│   ├── flink-console.sh
│   ├── flink-daemon.sh
│   ├── historyserver.sh
│   ├── jobmanager.sh
│   ├── kubernetes-jobmanager.sh
│   ├── kubernetes-session.sh
│   ├── kubernetes-taskmanager.sh
│   ├── pyflink-shell.sh
│   ├── sql-client.sh
│   ├── standalone-job.sh
│   ├── start-cluster.sh
│   ├── start-zookeeper-quorum.sh
│   ├── stop-cluster.sh
│   ├── stop-zookeeper-quorum.sh
│   ├── taskmanager.sh
│   ├── yarn-session.sh
│   └── zookeeper.sh
├── conf
│   ├── flink-conf.yaml
│   ├── log4j-cli.properties
│   ├── log4j-console.properties
│   ├── log4j.properties
│   ├── log4j-session.properties
│   ├── logback-console.xml
│   ├── logback-session.xml
│   ├── logback.xml
│   ├── masters
│   ├── workers
│   └── zoo.cfg
├── examples
│   ├── batch
│   │   ├── ConnectedComponents.jar
│   │   ├── DistCp.jar
│   │   ├── EnumTriangles.jar
│   │   ├── KMeans.jar
│   │   ├── PageRank.jar
│   │   ├── TransitiveClosure.jar
│   │   ├── WebLogAnalysis.jar
│   │   └── WordCount.jar
│   ├── gelly
│   │   └── flink-gelly-examples_2.12-1.15-SNAPSHOT.jar
│   ├── python
│   │   ├── datastream
│   │   │   ├── event_time_timer.py
│   │   │   ├── __init__.py
│   │   │   ├── process_json_data.py
│   │   │   ├── state_access.py
│   │   │   └── word_count.py
│   │   └── table
│   │       ├── __init__.py
│   │       ├── mixing_use_of_datastream_and_table.py
│   │       ├── multi_sink.py
│   │       ├── pandas
│   │       │   ├── conversion_from_dataframe.py
│   │       │   ├── __init__.py
│   │       │   └── pandas_udaf.py
│   │       ├── process_json_data.py
│   │       ├── process_json_data_with_udf.py
│   │       ├── windowing
│   │       │   ├── __init__.py
│   │       │   ├── over_window.py
│   │       │   ├── session_window.py
│   │       │   ├── sliding_window.py
│   │       │   └── tumble_window.py
│   │       └── word_count.py
│   ├── streaming
│   │   ├── Iteration.jar
│   │   ├── SessionWindowing.jar
│   │   ├── SocketWindowWordCount.jar
│   │   ├── StateMachineExample.jar
│   │   ├── TopSpeedWindowing.jar
│   │   ├── Twitter.jar
│   │   ├── WindowJoin.jar
│   │   └── WordCount.jar
│   └── table
│       ├── AdvancedFunctionsExample.jar
│       ├── ChangelogSocketExample.jar
│       ├── GettingStartedExample.jar
│       ├── StreamSQLExample.jar
│       ├── StreamWindowSQLExample.jar
│       ├── UpdatingTopCityExample.jar
│       └── WordCountSQLExample.jar
├── lib
│   ├── flink-connector-files-1.15-SNAPSHOT.jar
│   ├── flink-csv-1.15-SNAPSHOT.jar
│   ├── flink-dist-1.15-SNAPSHOT.jar
│   ├── flink-json-1.15-SNAPSHOT.jar
│   ├── flink-scala_2.12-1.15-SNAPSHOT.jar
│   ├── flink-shaded-zookeeper-3.4.14.jar
│   ├── flink-table_2.12-1.15-SNAPSHOT.jar
│   ├── log4j-1.2-api-2.14.1.jar
│   ├── log4j-api-2.14.1.jar
│   ├── log4j-core-2.14.1.jar
│   └── log4j-slf4j-impl-2.14.1.jar
├── LICENSE
├── log
│   ├── flink-slinkydeveloper-sql-client-fedora.log
│   ├── flink-slinkydeveloper-standalonesession-0-fedora.log
│   ├── flink-slinkydeveloper-standalonesession-0-fedora.out
│   ├── flink-slinkydeveloper-taskexecutor-0-fedora.log
│   └── flink-slinkydeveloper-taskexecutor-0-fedora.out
├── opt
│   ├── flink-azure-fs-hadoop-1.15-SNAPSHOT.jar
│   ├── flink-cep-1.15-SNAPSHOT.jar
│   ├── flink-cep-scala_2.12-1.15-SNAPSHOT.jar
│   ├── flink-gelly-1.15-SNAPSHOT.jar
│   ├── flink-gelly-scala_2.12-1.15-SNAPSHOT.jar
│   ├── flink-oss-fs-hadoop-1.15-SNAPSHOT.jar
│   ├── flink-python_2.12-1.15-SNAPSHOT.jar
│   ├── flink-queryable-state-runtime-1.15-SNAPSHOT.jar
│   ├── flink-s3-fs-hadoop-1.15-SNAPSHOT.jar
│   ├── flink-s3-fs-presto-1.15-SNAPSHOT.jar
│   ├── flink-shaded-netty-tcnative-dynamic-2.0.39.Final-14.0.jar
│   ├── flink-shaded-zookeeper-3.5.9.jar
│   ├── flink-sql-client_2.12-1.15-SNAPSHOT.jar
│   ├── flink-state-processor-api-1.15-SNAPSHOT.jar
│   └── python
│       ├── cloudpickle-1.2.2-src.zip
│       ├── py4j-0.10.8.1-src.zip
│       └── pyflink.zip
├── plugins
│   ├── external-resource-gpu
│   │   ├── flink-external-resource-gpu-1.15-SNAPSHOT.jar
│   │   ├── gpu-discovery-common.sh
│   │   └── nvidia-gpu-discovery.sh
│   ├── metrics-datadog
│   │   └── flink-metrics-datadog-1.15-SNAPSHOT.jar
│   ├── metrics-graphite
│   │   └── flink-metrics-graphite-1.15-SNAPSHOT.jar
│   ├── metrics-influx
│   │   └── flink-metrics-influxdb-1.15-SNAPSHOT.jar
│   ├── metrics-jmx
│   │   └── flink-metrics-jmx-1.15-SNAPSHOT.jar
│   ├── metrics-prometheus
│   │   └── flink-metrics-prometheus-1.15-SNAPSHOT.jar
│   ├── metrics-slf4j
│   │   └── flink-metrics-slf4j-1.15-SNAPSHOT.jar
│   ├── metrics-statsd
│   │   └── flink-metrics-statsd-1.15-SNAPSHOT.jar
│   └── README.txt
└── README.txt
```